builder/bundleset.go

```diff
@@ -98,6 +98,16 @@ func sortBundles(bundles bundleSet) ([]*bundle, error) {
 					return err
 				}
 			}
+			for _, opt := range b.OptionalIncludes {
+				bundle, exists := bundles[opt]
+				if !exists {
+					return fmt.Errorf("bundle %q includes optional (also-add) bundle %q which is not available", b.Name, opt)
+				}
+				err := visit(bundle)
```

Comment:
I don't believe we need to visit the optional bundles. They will not be used to determine the contents of the manifest that includes them and since they are an independent bundles, they will be validated on their own in a separate step.

Reply:
if we remove this part we won't be able to detect cycle errors in also-add, because of this, this test fail TestParseBundleSet (https://github.com/clearlinux/mixer-tools/pull/580/files#diff-a579009ef8fb588be3dec39355f8b2f5R451)

Reply:
As we discussed on IRC, loops are OK with also-add since there is no dependency among them.

Reply:
I removed the part where we were visiting the optional bundles but left the part where it checks for existence of the optional bundle